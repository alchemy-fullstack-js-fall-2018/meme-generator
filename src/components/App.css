@import url('https://fonts.googleapis.com/css?family=Indie+Flower|Kumar+One+Outline|Open+Sans|Roboto');

.serif {
  font-family: Times, , Georgia, serif;
}

.sans-serif {
  font-family: Verdana, Arial, Helvetica, sans-serif;
}

.monospace {
  font-family: Lucida, Courier, monospace;
}

.cursive {
  font-family: cursive;
}

.fantasy {
  font-family: fantasy;
}
.helvetica {
  font-family: 'Helvetica'
}
.open-sans {
  font-family: 'Open Sans', sans-serif;
}

.form {
  display: block;
}

.meme {
  position: relative;
  text-align: center;
  color: black;
  margin: 0px;
  width: auto;
  height: auto;
}

.image {
  object-fit: cover;
  object-position: left;
  z-index: -1;
}

.text {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  font-size: 55px;
  font-weight: bold;
  color: white;
}

.header {
    position: absolute;
    top: 30px;
    left: 50%;
    transform: translate(-50%, -50%);
}

.footer {
    position: absolute;
    bottom: -30px;
    left: 50%;
    transform: translate(-50%, -50%);
}