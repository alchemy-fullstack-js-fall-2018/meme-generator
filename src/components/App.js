import React, { Component, Fragment } from 'react';
import styles from './App.css';
import domToImage from 'dom-to-image';
import fileSaver from 'file-saver';
export default class App extends Component {

  state = {
    url: 'http://static.origos.hu/s/img/i/1801/20180108doge-dogecoin-kriptovaluta.jpg',
    header: 'Much',
    footer: 'Disappoint',
    font: 'fantasy',
    color: { color: 'white' }
  };

  handleChosenFile = ({ target }) => {
    let url = window.URL.createObjectURL(target.files[0]);
    this.setState({ url });
  };

  onChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };
  changeColor = ({ target }) => {
    this.setState({ [target.name]:{ [target.name]: target.value } });
  };

  saveMeme = (event) => {
    event.preventDefault();
    domToImage.toPng(document.getElementById('meme'))
      .then(meme => fileSaver.saveAs(meme));
  };

  render() {
    const { url, header, footer, font, color } = this.state;
    const options = ['helvetica', 'open-sans', 'serif', 'monospace', 'cursive', 'fantasy'].map(font => {
      return <option key={font} value={font}>{font}</option>;
    });

    const colorOptions = ['White', 'Yellow', 'Blue', 'Pink', 'Purple', 'Red'].map(color => {
      return <option key={color} value={color}>{color}</option>;
    });
    return (
      <Fragment>
        <h1>Lets Make Some Memes!</h1>
        <form onSubmit={this.saveMeme} className="form">
          <label htmlFor="url">Paste a Url:</label><br/>
          <input name="url" placeholder="Add an Image URL" value={url} onChange={this.onChange}/><br/>
          <p>OR</p>
          <label htmlFor="file">Upload an Image:</label><br/>
          <input name="file" type="file" onChange={this.handleChosenFile}/><br/>
          <input name="header" placeholder="Add Header" value={header} onChange={this.onChange}/><br/>
          <input name="footer" placeholder="Add Footer" value={footer} onChange={this.onChange}/><br/>
          <label htmlFor="font">Font Type:</label>
          <select name="font" defaultValue={font} onChange={this.onChange}>
            {options}
          </select><br/>
          <label htmlFor="color">Color Type:</label>
          <select name="color" defaultValue={color} onChange={this.changeColor}>
            {colorOptions}
          </select><br/>
          <button type="submit">Save This Meme</button>
        </form>
        <div id="meme" className={styles.meme}>
          <img id="img"src={url} alt="meme" className={styles.image}/>
          <div className={styles.text}>
            <div className={styles.header}>
              <span className={styles[font]} style={color}>{header.toUpperCase()}</span>
            </div>
            <div className={styles.footer}>
              <span className={styles[font]} style={color}>{footer.toUpperCase()}</span>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}
