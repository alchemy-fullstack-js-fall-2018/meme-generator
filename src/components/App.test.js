import React from 'react';
import App from './App';
import render from 'react-test-renderer';

describe('App Component', () => {
  test('header snapshot', () => {
    expect(render
      .create(<App />)
      .toJSON()).toMatchSnapshot();
  });
});
