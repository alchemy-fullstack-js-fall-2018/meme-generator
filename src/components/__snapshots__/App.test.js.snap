// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component header snapshot 1`] = `
Array [
  <h1>
    Lets Make Some Memes!
  </h1>,
  <form
    className="form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="url"
    >
      Paste a Url:
    </label>
    <br />
    <input
      name="url"
      onChange={[Function]}
      placeholder="Add an Image URL"
      value="http://static.origos.hu/s/img/i/1801/20180108doge-dogecoin-kriptovaluta.jpg"
    />
    <br />
    <p>
      OR
    </p>
    <label
      htmlFor="file"
    >
      Upload an Image:
    </label>
    <br />
    <input
      name="file"
      onChange={[Function]}
      type="file"
    />
    <br />
    <input
      name="header"
      onChange={[Function]}
      placeholder="Add Header"
      value="Much"
    />
    <br />
    <input
      name="footer"
      onChange={[Function]}
      placeholder="Add Footer"
      value="Disappoint"
    />
    <br />
    <label
      htmlFor="font"
    >
      Font Type:
    </label>
    <select
      defaultValue="fantasy"
      name="font"
      onChange={[Function]}
    >
      <option
        value="helvetica"
      >
        helvetica
      </option>
      <option
        value="open-sans"
      >
        open-sans
      </option>
      <option
        value="serif"
      >
        serif
      </option>
      <option
        value="monospace"
      >
        monospace
      </option>
      <option
        value="cursive"
      >
        cursive
      </option>
      <option
        value="fantasy"
      >
        fantasy
      </option>
    </select>
    <br />
    <label
      htmlFor="color"
    >
      Color Type:
    </label>
    <select
      defaultValue={
        Object {
          "color": "white",
        }
      }
      name="color"
      onChange={[Function]}
    >
      <option
        value="White"
      >
        White
      </option>
      <option
        value="Yellow"
      >
        Yellow
      </option>
      <option
        value="Blue"
      >
        Blue
      </option>
      <option
        value="Pink"
      >
        Pink
      </option>
      <option
        value="Purple"
      >
        Purple
      </option>
      <option
        value="Red"
      >
        Red
      </option>
    </select>
    <br />
    <button
      type="submit"
    >
      Save This Meme
    </button>
  </form>,
  <div
    className="meme"
    id="meme"
  >
    <img
      alt="meme"
      className="image"
      id="img"
      src="http://static.origos.hu/s/img/i/1801/20180108doge-dogecoin-kriptovaluta.jpg"
    />
    <div
      className="text"
    >
      <div
        className="header"
      >
        <span
          className="fantasy"
          style={
            Object {
              "color": "white",
            }
          }
        >
          MUCH
        </span>
      </div>
      <div
        className="footer"
      >
        <span
          className="fantasy"
          style={
            Object {
              "color": "white",
            }
          }
        >
          DISAPPOINT
        </span>
      </div>
    </div>
  </div>,
]
`;
